container/ubuntu16_04/layers/bazel/BUILD

```diff
@@ -26,22 +29,39 @@ language_tool_layer(
     name = "bazel-tools",
     base = "@ubuntu16_04//image",
     packages = [
+        "bash-completion",
+        "binutils",
         "patch",
         "unzip",
         "zip",
     ],
 )
 
-language_tool_layer(
-    name = "bazel-ltl",
-    additional_repos = [
-        "deb [arch=amd64] http://storage.googleapis.com/bazel-apt stable jdk1.8",
-    ],
+toolchain_container(
+    name = "bazel_installer_image",
     base = "@ubuntu16_04//image",
-    keys = [
-        "@bazel_gpg//file",
+    files = [
+        "@bazel_installer//file",
     ],
     packages = [
-        "bazel",
+        "unzip",
+    ],
+)
+
+container_run_and_extract(
```

Comment:
this is pretty sweet. Its using docker run under the hood or something else? Although its not ideal if it runs docker I think it should be fine. 
We should try to add a test to verify the output of this rule is deterministic (if not trivial, otherwise leave for next cl)

Reply:
It is using a docker run command.

I'll probably leave that for another CL.